File: task310-rest/task320-jpa 2/src/main/java/com/example/task320jpa/repository/NoteRepository.java
package com.example.task320jpa.repository;

import com.example.task320jpa.entity.Note;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

/**
 * JPA Repository для сущности Note
 */
@Repository
public interface NoteRepository extends JpaRepository<Note, Long> {
    
    /**
     * Найти все заметки для твита
     */
    List<Note> findByTweetId(Long tweetId);
    
    /**
     * Найти все заметки для твита с пагинацией
     */
    Page<Note> findByTweetId(Long tweetId, Pageable pageable);
    
    /**
     * Найти все заметки с пагинацией
     */
    Page<Note> findAll(Pageable pageable);
}

File: 351002/Abrashin/src/main/kotlin/com/example/distcomp/repository/NoteRepository.kt
package com.example.distcomp.repository

import com.example.distcomp.model.Note
import org.springframework.stereotype.Repository

@Repository
class NoteRepository : InMemoryRepository<Note>() {
    fun findByTweetId(tweetId: Long): List<Note> = storage.values.filter { it.tweetId == tweetId }
}
